internal/provider/resource_custom_location.go

```diff
@@ -0,0 +1,219 @@
+				Computed:    true,
+			},
+		},
+		Importer: &schema.ResourceImporter{},
+	}
+}
+
+func resourceCustomLocationCreate(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+
+	c := m.(*client.Client)
+
+	// Define & Build
+	locationName := d.Get("name").(string)
+	isLocationEnabled := d.Get("enabled").(bool)
+	description := GetString(d.Get("description").(string))
+	tags := GetStringMap(d.Get("tags"))
+	provider := d.Get("cloud_provider").(string)
+
+	if provider != "byok" {
+		return diag.Errorf("provider must be byok for custom locations")
+	}
+
+	location := &client.Location{}
+	location.Name = &locationName
+	location.Spec = &client.LocationSpec{}
+	location.Spec.Enabled = GetBool(isLocationEnabled)
+	location.Provider = &provider
+	location.Tags = tags
+	location.Description = description
+	location.Status = nil
+	createdLocation, _, err := c.CreateCustomLocation(*location)
+
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	return setCustomLocationResource(d, createdLocation)
+}
+
+func resourceCustomLocationRead(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+
+	c := m.(*client.Client)
+
+	location, code, err := c.GetLocation(d.Id())
+
+	if code == 404 {
+		d.SetId("")
+		return nil
+	}
+
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	return setCustomLocationResource(d, location)
+}
+
+func resourceCustomLocationUpdate(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+	if d.HasChanges("tags", "enabled", "description") {
+		c := m.(*client.Client)
+
+		// Define & Build
+		locationToUpdate := client.Location{
+			Spec: &client.LocationSpec{
+				Enabled: GetBool(d.Get("enabled").(bool)),
+			},
+		}
+
+		locationToUpdate.Name = GetString(d.Get("name").(string))
+
+		if d.HasChange("tags") {
+			locationToUpdate.Tags = GetTagChanges(d)
+		}
+
+		if d.HasChange("description") {
+			locationToUpdate.Description = GetString(d.Get("description").(string))
+		}
+
+		// Update
+		updatedLocation, _, err := c.UpdateLocation(locationToUpdate)
+
+		if err != nil {
+			return diag.FromErr(err)
+		}
+
+		return setCustomLocationResource(d, updatedLocation)
+	}
+
+	return nil
+}
+
+func resourceCustomLocationDelete(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+
+	c := m.(*client.Client)
+
+	locationName := d.Get("name").(string)
+
+	// Get location
+	location, _, err := c.GetLocation(locationName)
+
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	// Handle nil just to be on the safe side
+	if location.Spec == nil {
+		location.Spec = &client.LocationSpec{}
+	}
+
+	// Set enabled to its default value
+	location.Spec.Enabled = GetBool(true)
+
+	// Remove status
+	location.Status = nil
+
+	// Send an update request
+	_, _, err = c.UpdateLocation(*location)
+
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	d.SetId("")
+	return nil
+}
```

Comment:
This is an incorrect implementation. We need to delete the custom location, not update it. Look at the following implementation as an example for destroying a resource:
https://github.com/controlplane-com/terraform-provider-cpln/blob/main/internal/provider/resource_gvc.go#L273